core_types/src/agent.rs

```diff
@@ -42,6 +42,21 @@ impl KeyBuffer {
         KeyBuffer(b.clone())
     }
 
+    /// generate a key buffer from raw bytes from two parts (no correction)
+    pub fn with_raw_parts(
+        a: &[u8; KeyBuffer::HALF_KEY_LEN],
+        b: &[u8; KeyBuffer::HALF_KEY_LEN],
+    ) -> KeyBuffer {
+        let mut buf: [u8; KeyBuffer::KEY_LEN] = [0; 64];
+        for i in 0..KeyBuffer::HALF_KEY_LEN {
+            buf[i] = a[i];
+        }
+        for i in 0..KeyBuffer::HALF_KEY_LEN {
+            buf[KeyBuffer::HALF_KEY_LEN + i] = b[i];
+        }
```

Comment:
would be awesome if we could use this: https://doc.rust-lang.org/1.3.0/std/slice/bytes/fn.copy_memory.html

not sure * how * unstable that rust api is...

Reply:
I just tried this out. But the copy_memory code in rust still needs review. So I don't think I can access the bytes module. 
Also, I need to concat two [u8]s here. So I could use this instead of the first loop. but I would still need the second loop

Reply:
alas... if only rust were more mature.